### PR TITLE
Add option for sparse matrix parsing when creating CDS

### PR DESCRIPTION
**Pull request by a-solovyev12**
This PR adds ability to read sparse matrices in .mtx format when creating CDS object.

### CODE REVIEW

File: install_monocle3.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@ conda config --set always_yes yes --set changeps1 no --set channel_priority stri
 
 conda $cmd -q -n $env_name python=$python_version louvain umap-learn \
     r-assertthat r-matrix r-matrix.utils r-ggplot2 r-ggrepel r-viridis r-dplyr r-tidyr r-stringr r-reshape2 \
-    r-plotly r-pryr r-base64enc r-hexbin r-crosstalk r-data.table r-codetools \
+    r-plotly r-pryr r-base64enc r-hexbin r-crosstalk r-data.table r-codetools r-optparse \
```
Comment (nh3) [inline install_monocle3.sh:26]: It's intriguing that it didn't require explicit mention of r-optparse in previous CI builds. Perhaps some dependencies have changed in conda. Anyway, good to be explicit.

**Comment by nh3** [comment]
Very good addition! Perhaps also mention this in the usage message? https://github.com/ebi-gene-expression-group/monocle-scripts/blob/b39b0f22fa3701ce31f7259ecf109de414ed3a90/R/cmd_args.R#L124

**Comment by nh3** [comment]
Would be great if the tests can be updated to use mtx input, since I imagine it will be the main input format when put to use.

**Comment by a-solovyev12** [comment]
@nh3 Updating the tests is definitely a good idea. Do you know if there is a link to the same dataset in .mtx format somewhere? Otherwise, I can write a function to extract the matrix from .rds object and stick it into downstream test cases.

**Comment by nh3** [comment]
I am not aware of an mtx version of the same dataset online. I think your solution is very reasonable.

**Review by nh3 — APPROVED** [review]
LGTM!

**Comment by a-solovyev12** [comment]
Thank you for review @nh3

**Comment by a-solovyev12** [comment]
Hi @nh3, could you please make a release with updated version? I'd like to incorporate this into a workflow I'm working on. Thanks!

**Comment by nh3** [comment]
Hi @a-solovyev12, I've given you permissions to release it on your own.

**Pull request merged**